lib/cocoapods/resolver.rb

```diff
@@ -132,7 +137,9 @@ def resolver_specs_by_target
               payload = vertex.payload
               non_library = !used_by_aggregate_target_by_spec_name.fetch(vertex.name)
               spec_source = payload.respond_to?(:spec_source) && payload.spec_source
-              ResolverSpecification.new(payload, non_library, spec_source)
+              transitive = !explicitly_included_by_aggregate_by_spec_name.fetch(vertex.name, false)
+              incoming_spec_vertices = vertex.incoming_edges.map { |edge| edge.origin.payload }
```

Comment:
does this need to filter by platform, potentially?

Reply:
ah potentially yes, I think if a specification has a platform specific dependency then that could be an issue. we support that correct? `s.ios.dependency 'SomePod'`